code/modules/reagents/reagents/reagents_drug.dm

```diff
@@ -254,12 +254,86 @@
 		M.hallucination += 10
 
 /datum/reagent/hyperzine/methamphetamine //slightly better than 'zine
-	name = "Methamphetamine" //Only used on the Laundromat spess vault
+	name = "Methamphetamine"
 	id = METHAMPHETAMINE
 	description = "It uses a different manufacture method but it is every bit as pure."
 	color = "#89CBF0" //baby blue
 	custom_metabolism = 0.01
 	overdose_am = 30
+	data = "no motor mouth"
+
+/datum/reagent/hyperzine/methamphetamine/on_mob_life(var/mob/living/M)
+	if(..())
+		return 1
+//most of this code purloined from piccolyn 
+	if(M && prob(5)) //ocassionally switches you to HARM intent
+		M.a_intent = I_HURT
+		if(M?.hud_used?.action_intent)
+			M.hud_used.action_intent.icon_state = "intent_hurt"
+	if(prob(5) && M.stat == CONSCIOUS)
+		M.emote(pick("twitch","blink_r","shiver")) 
+	if(M && prob(5)) //you will occasionally say something completely out of left field to a seccie, centcommie, captain or HoP
+		var/list/nearest_officer = null
+		for(var/mob/living/L in view(M))
+			if(L == M)
+				continue
+			if(L.stat)
+				continue
+			if(nearest_officer && get_dist(L,M)>=get_dist(nearest_officer,M))
+				continue //We already have a closer living target
+			if(ishuman(L))
+				var/mob/living/carbon/human/H = L
+				var/list/seccie_uniforms_list = list(/obj/item/clothing/under/rank/secformal,
+														/obj/item/clothing/under/rank/warden,
+														/obj/item/clothing/under/rank/security,
+														/obj/item/clothing/under/rank/security2,
+														/obj/item/clothing/under/rank/collar,
+														/obj/item/clothing/under/det,
+														/obj/item/clothing/under/rank/head_of_security,
+														/obj/item/clothing/under/rank/centcom,
+														/obj/item/clothing/under/rank/metrocop,
+														/obj/item/clothing/under/rank/head_of_personnel,
+														/obj/item/clothing/under/rank/captain)
+				if(H.is_wearing_any(seccie_uniforms_list,slot_w_uniform))
+					//Check to see if it's wearing the right stuff
+					nearest_officer = H
+			else if(isrobot(L))
+				var/mob/living/silicon/robot/R = L
+				if(HAS_MODULE_QUIRK(R, MODULE_IS_THE_LAW))
+					nearest_officer = R
+		if(!nearest_officer)
+			return 1
+		var/D = "officer"
+		if(ishuman(nearest_officer))
+			var/mob/living/carbon/human/H = nearest_officer
+			D = get_first_word(H.name)
```

Comment:
Is the name updated by identity-changing things such as an ID while wearing a gas mask?

Reply:
it's literally the same call as the piccolyn one but with the vars renamed to not be confusing, so if it works for piccolyn, then it'd work for this too.